### PR TITLE
Prop that displays a search bar to filter facets

### PR DESCRIPTION
**Pull request by thalytafabrine**
#### What problem is this solving?
allows users to filter the options of a filter when it has many options

#### How to test it?

[Workspace](https://facetssearch--storecomponents.myvtex.com/apparel---accessories)

#### Screenshots or example usage:

searchbar

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -229,6 +229,7 @@ Structure the `search-result-layout` or the `search-result-layout.customQuery`,
 | `blockClass`        | `string`       | Unique block ID to be used in [CSS customization](https://vtex.io/docs/recipes/style/using-css-handles-for-store-customization#using-the-blockclass-property)                                                                                    | `undefined`              |
 | `trackingId` | `string` | ID to be used in Google Analytics to track store metrics based on the Search Result block. |  `Search result` | 
 | `mobileLayout`      | `Object` | Controls how the search results page will be displayed to users when using the mobile layout. Possible props and their respective values can be found below.                                                                                                                | `undefined`              |
+| `showFacetSearch` | `number` | Minimum number of options that must exist in a filter for the search bar to be displayed. If it is `0`, the search bar will always be displayed. |  `undefined` |
```
Comment (klzns) [inline docs/README.md:232]: The name of this prop suggests that it's a boolean value.

Maybe we can change to something like `searchFacetMinimum`? Open to other names

File: react/components/SearchFilterBar.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  name: string
+  handleChange: (value: string) => void
+}
+
+export const SearchFilterBar = ({ name, handleChange }: Props) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  const intl = useIntl()
+
+  return (
+    <div className={classNames('mb3', handles.searchFilterBar)}>
```
Comment (klzns) [inline react/components/SearchFilterBar.tsx:26]:
```suggestion
    <div className={`${handles.searchFilterBar} mb3`}>
```

File: react/components/SearchFilterBar.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { Input } from 'vtex.styleguide'
+import { useCssHandles } from 'vtex.css-handles'
+import { defineMessages, useIntl } from 'react-intl'
+import classNames from 'classnames'
```
Comment (klzns) [inline react/components/SearchFilterBar.tsx:5]:
```suggestion
```

File: docs/README.md (hunk 1 above)
Comment (thalytafabrine) [reply docs/README.md:232]: what do you think about `minimumSearchThreshold` or something like this?

Comment (klzns) [reply docs/README.md:232]: Like it!

Comment (klzns) [reply docs/README.md:232]: Or `thresholdForFacetSearch`

I think the word facet is important here

File: react/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+ "compilerOptions": {
+   "jsx": "react",
+   "esModuleInterop": true
+ } 
+}
```
Comment (klzns) [inline react/tsconfig.json:6]: Change to use the default config

https://github.com/vtex-apps/react-app-template/blob/master/react/tsconfig.json

File: docs/README.md [hunk 5]
```diff
@@ -229,6 +229,7 @@ Structure the `search-result-layout` or the `search-result-layout.customQuery`,
 | `blockClass`        | `string`       | Unique block ID to be used in [CSS customization](https://vtex.io/docs/recipes/style/using-css-handles-for-store-customization#using-the-blockclass-property)                                                                                    | `undefined`              |
 | `trackingId` | `string` | ID to be used in Google Analytics to track store metrics based on the Search Result block. |  `Search result` | 
 | `mobileLayout`      | `Object` | Controls how the search results page will be displayed to users when using the mobile layout. Possible props and their respective values can be found below.                                                                                                                | `undefined`              |
+| `thresholdForFacetSearch` | `number` | Minimum number of options that must exist in a filter for the search bar to be displayed. If it is `0`, the search bar will always be displayed. |  `undefined` |
```
Comment (guerreirobeatriz) [inline docs/README.md:232]:
```suggestion
| `thresholdForFacetSearch` | `number` | Minimum number of filters that must be displayed on the interface for a search bar to be displayed. If you declare `0`, the search bar will always be displayed. |  `undefined` | 
```

**Review by klzns — APPROVED** [review]

**Pull request merged**